src/main/java/org/truffleruby/extra/ffi/PointerNodes.java

```diff
@@ -70,6 +70,54 @@ public DynamicObject allocate(DynamicObject pointerClass) {
 
     }
 
+    @Primitive(name = "pointer_find_type_size", needsSelf = false)
+    public static abstract class PointerFindTypeSizePrimitiveNode extends PrimitiveArrayArgumentsNode {
+
+        @TruffleBoundary
+        @Specialization(guards = "isRubySymbol(type)")
```

Comment:
I don't know how frequently this is called. Would it make sense to cache the lookup?

Reply:
It's called for e.g.: `MemoryPointer.new(:int, 4)`. I don't think it's performance critical. We'd need to split the `.new` to make the cache effective.